src/Microsoft.Health.Fhir.Core/Features/Conformance/GetSmartConfigurationHandler.cs

```diff
@@ -0,0 +1,79 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Hl7.Fhir.ElementModel;
+using Hl7.Fhir.Serialization;
+using MediatR;
+using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.Fhir.Core.Extensions;
+using Microsoft.Health.Fhir.Core.Features.Conformance.Models;
+using Microsoft.Health.Fhir.Core.Features.Operations;
+using Microsoft.Health.Fhir.Core.Messages.Get;
+using Microsoft.Health.Fhir.Core.Models;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Core.Features.Conformance
+{
+    public class GetSmartConfigurationHandler : IRequestHandler<GetSmartConfigurationRequest, GetSmartConfigurationResponse>
+    {
+        private readonly SecurityConfiguration _securityConfiguration;
+        private readonly IModelInfoProvider _modelInfoProider;
+
+        public GetSmartConfigurationHandler(SecurityConfiguration securityConfiguration, IModelInfoProvider modelInfoProvider)
+        {
+            EnsureArg.IsNotNull(securityConfiguration, nameof(securityConfiguration));
+            EnsureArg.IsNotNull(modelInfoProvider, nameof(modelInfoProvider));
+
+            _securityConfiguration = securityConfiguration;
+            _modelInfoProider = modelInfoProvider;
+        }
+
+        public Task<GetSmartConfigurationResponse> Handle(GetSmartConfigurationRequest request, CancellationToken cancellationToken)
+        {
+            return Task.FromResult(Handle(request));
+        }
+
+        protected GetSmartConfigurationResponse Handle(GetSmartConfigurationRequest request)
+        {
+            EnsureArg.IsNotNull(request, nameof(request));
+
+            if (_securityConfiguration.Authorization.Enabled)
+            {
+                string baseEndpoint = _securityConfiguration.Authentication.Authority;
+
+                try
+                {
+                    Uri authorizationEndpoint = new Uri(string.Join(baseEndpoint, "/authorize"));
+                    Uri tokenEndpoint = new Uri(string.Join(baseEndpoint, "/token"));
+                    ICollection<string> capabilities = new List<string>
+                    {
+                        "launch-standalone",
```

Comment:
Do we want to advertise capabilities we don't support? I worry this will cause confusion for customers.

Reply:
This list was constructed to make the Inferno test for capabilities pass (from this [chat.fhir thread](https://chat.fhir.org/#narrow/stream/179309-inferno/topic/well-known.20.22capabilities.22), it's mentioned that the scopes listed [here](https://github.com/onc-healthit/onc-certification-g10-test-kit/blob/7c55ebc4edf215ac00ca585f5b77ad4673936b2a/lib/onc_certification_g10_test_kit/smart_standalone_patient_app_group.rb#L65-L73) are likely the ones that Inferno is checking to see are present).

Also, note that although `launch-standalone`, `client-public`, `client-confidential-symmetric`, `patient-context` are not supported by the FHIR Service without APIM on top, this list will be returned only if security configuration authorization is enabled [per line 43](https://github.com/microsoft/fhir-server/pull/2887/files#diff-ea5026642fbc21f7ceae4224a955bed92513802f1209740b2a502014377c64b9R43).

I agree that this can misrepresent what FHIR server supports, but it looks like we'll need them for Inferno testing.. thoughts?

Reply:
I agree that the FHIR server should advertise only what it supports.  If additional capabilities are added by APIM/Function, then APIM can update the .well-known response with those capabilities.

Reply:
Makes sense. Removing `launch-standalone`, `client-public`, `client-confidential-symmetric`, `context-standalone-patient`.